connector/oidc/oidc.go

```diff
@@ -286,6 +291,11 @@ func (c *oidcConnector) createIdentity(ctx context.Context, identity connector.I
 	}
 	hostedDomain, _ := claims["hd"].(string)
 
+	if c.preferredUsernameKey == "" {
+		c.preferredUsernameKey = "username"
```

Comment:
Shouldn't this be `preferred_username`?

Reply:
`preferred_username` is meant for `Shorthand name` according to OIDC spec. The PR here is aiming for `username` claim, which is usualy unique. Other connectors like github and gitlab has similar implementations

https://github.com/dexidp/dex/blob/aca67b083984982c11942ac505b883c1d75bf6f9/connector/gitlab/gitlab.go#L152

https://github.com/dexidp/dex/blob/aca67b083984982c11942ac505b883c1d75bf6f9/connector/github/github.go#L269

Reply:
But `preferred_username` is actually a standard claim, so I would assume the OIDC coonnector would return that (given the underlying connector supports it).

Having an arbitrary, non-standard claim as the default value is confusing at best.